### PR TITLE
[UPDATE] organize display of select items

### PR DESCRIPTION
**Pull request by georgehristov**
Update code for select items display:
- use description callback if available
- move tab caption routine to a single method `call_select_item_format_callback`

### CODE REVIEW

**Comment by bukowskiadam** [comment]
With your commit I get additional tab caption in some places. Is this indented behavior?
This comes from 2388 line.

**Comment by georgehristov** [comment]
No, it is not intended. It is actually the reason I started this update.
`autoselect_label` calls `create_default_linked_label` with `$table_name` parameter set to false exactly for that reason.
Can you tell me which module this double tab caption appears in if it is a core module?
If not please let me know the context so I can simulate. I presume it is with a custom format callback for the select option.

**Comment by bukowskiadam** [comment]
It's not doubled. Sometimes it's not required (for instance checking company name unique), but those places should be fixed by passing `$table_name=false`.
The other place is shoutbox, where you can link to any record with bb tag `[rb=contact/1]`
In fact it looks like a good change :)

**Comment by georgehristov** [comment]
OK, if it is good it is good :)
What I like to re-organize here is this `$args` argument of the `autoselect_label` and `call_select_item_format_callback` methods.

Is this argument fully used in any premium or custom module or I can streamline the code a bit more in regard to the select items formatting?

**Comment by bukowskiadam** [comment]
I'm not sure right now, but I don't see any text occurrences of `autoselect_label` and `call_select_item_format_callback` in any premium module.

**Comment by georgehristov** [comment]
Actually I added `call_select_item_format_callback` recently and `autoselect_label` is the default callback.
Where this `$args` might be used is item format callbacks of select fields (returned by the advanced_param_callback - the 3rd setting in the select `param`)

**Comment by bukowskiadam** [comment]
Well... maybe leave it as is. Really I wish to make the release ASAP, but your changes are good, so I've got the problem of merging and testing them :wink:

**Comment by georgehristov** [comment]
I will make the changes I have in mind but keep it for a pull request after the release. Hasty changes before release are not good.

**Comment by georgehristov** [comment]
I added a small modification for the `display_select` method to use description callback if available.

**Comment by georgehristov** [comment]
I have never used bbcode although I implemented functionality which can easily be done with bbcode in one custom module.
Now somehow could not make it work as I could not create an expresion that matches the regex in `Utils_BBCodeCommon::parse`. Any hint?

File: modules/Utils/RecordBrowser/RecordBrowserCommon_0.php [hunk 1]
```diff
@@ -165,15 +165,16 @@ public static function display_select($record, $nolink=false, $desc=null, $tab=n
...
-    					list($select_tab, $v) = explode('/', $v, 2);
+    				$tab_id = self::decode_record_token($v, $param['single_tab']);
     				
-    				if($param['cols']) {
-    					$res = self::create_linked_label($select_tab, $param['cols'], $v, $nolink);
+    				if (!$tab_id) continue;
+    				 
+    				list($select_tab, $id) = $tab_id;
+
+    				if(self::get_description_callback($tab) || empty($param['cols'])) {
```
Comment (bukowskiadam) [inline modules/Utils/RecordBrowser/RecordBrowserCommon_0.php]: This change makes different behavior and I would expect previous one.
If you've got description callback set for certain record type (e.g. company), then you won't be able to use custom label using other columns. For instance you'd like to create select field, where field xxx will be shown, but it's not in the description callback.

**Comment by bukowskiadam** [comment]
And about the bbcode - you have to register your handler with `Utils_BBCodeCommon::new_bbcode`. `rb` code is built-in (now I don't remember why it's not the registered one, but I've got some issues in the past)

**Comment by georgehristov** [comment]
I have updated the commits now as per your last - first listed cols are used and then description callback.
Functionality is same as before my changes, no the only difference is that getting description callback is in a separate method with cache.
I will not delay more with the version release :wink:

**Pull request merged**